decompiler/pipeline/controlflowanalysis/expression_simplification/constant_folding.py

```diff
@@ -133,8 +133,9 @@ def _constant_fold_shift(constants: list[Constant], fun: Callable[[int, int], in
         raise IncompatibleOperandCount("Expected exactly 2 constants to fold")
 
     left, right = constants
+    operand_signed = isinstance(left.type, Integer) and left.type.signed
 
-    return fun(normalize_int(left.value, left.type.size, left.type.signed and signed), right.value)
+    return fun(normalize_int(left.value, left.type.size, operand_signed and signed), right.value)
```

Comment:
I know it is a very minor, nitpicking thing to ask, but could you please change the order the parameters get evaluated in? :D When the `signed` parameter is set to false, we have no buisness checking the operand

so I'd propse something like this:
```python
operand_signed = signed and isinstance(left.type, Integer) and left.type.signed
```

Reply:
Changed with new commit.